### PR TITLE
test old trashbin api by restoring deleted files using the webUi and the checking the activity

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

webui acceptance test to restore file/folder using the webui and checking the activity

## Related Issue
Fixed: #778 

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/webUIActivityDeleteRestore/restore.feature [hunk 1]
```diff
@@ -284,3 +284,43 @@ Feature: Restored files/folders activities
     Then the activity number 1 should contain message "User One restored lorem.txt" in the activity page
     And the activity number 2 should contain message "User One deleted lorem.txt" in the activity page
     And the activity number 3 should have a message saying that you have shared folder "simple-folder" with user "User One"
+
+  Scenario Outline: Restore a file using the webUI and check the activity
```
Comment (phil-davis) [inline tests/acceptance/features/webUIActivityDeleteRestore/restore.feature]: These look good for doing a quick check that the webUI trashbin restore actions are still triggering activity events.
Put them in a separate feature file - for example `restoreUsingWebUI.feature` 
And in that feature file do not put this line in the `Background` section:
```
Given the administrator has enabled DAV tech_preview
```
That way, these scenarios will definitely run without the new API, so we also will be sure that the new API is not accidentally used.

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/activity/794/24/10
For batch restore, the activity list of files restored is in a different order in Firefox.

https://drone.owncloud.com/owncloud/activity/794/23/10
On chrome it reran the scenario and got the expected answer the 2nd time!

I guess the order that the file names are listed in the activity entry does not matter. So instead of step:
```
Then the activity number 1 should have message "You restored lorem.txt, strängé नेपाली folder and 'single'quotes" in the activity page
```
there needs to be a "smarter" step that checks that the message has the file names, but in any order.

File: tests/acceptance/features/bootstrap/WebUIActivityContext.php [hunk 2]
```diff
@@ -326,6 +327,35 @@ public function theActivityNumberShouldHaveAMessageSayingThatUserHasSharedWithYo
 		PHPUnit\Framework\Assert::assertEquals($message, $latestActivityMessage);
 	}
 
+	/**
+	 * @Then the activity number :arg1 should have a message saying that you restored the following files in the activity page:
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/WebUIActivityContext.php]:
```suggestion
	 * @Then the activity number :index should have a message saying that you restored the following files in the activity page:
```

File: tests/acceptance/features/webUIActivityDeleteRestore/restoreUsingWebUI.feature [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  So that I know what happened in my cloud storage
+
+  Background:
+    Given user "user0" has been created with default attributes and skeleton files
+    And user "user0" has logged in using the webUI
+
+  Scenario Outline: Restore a file using the webUI and check the activity
+    Given user "user0" has deleted file "<file>"
+    And the user has browsed to the trashbin page
+    When the user restores file "lorem.txt" from the trashbin using the webUI
```
Comment (skshetry) [inline tests/acceptance/features/webUIActivityDeleteRestore/restoreUsingWebUI.feature]: Should not this be `"<file>"`?

```suggestion
    When the user restores file "<file>" from the trashbin using the webUI
```

File: tests/acceptance/features/webUIActivityDeleteRestore/restoreUsingWebUI.feature [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    And the user browses to the activity page
+    Then the activity number 1 should have message "You restored lorem.txt" in the activity page
```
Comment (skshetry) [inline tests/acceptance/features/webUIActivityDeleteRestore/restoreUsingWebUI.feature]: Same here. 
```suggestion
    Then the activity number 1 should have message "You restored <file>" in the activity page
```

File: tests/acceptance/features/bootstrap/WebUIActivityContext.php [hunk 5]
```diff
@@ -326,6 +327,35 @@ public function theActivityNumberShouldHaveAMessageSayingThatUserHasSharedWithYo
...
+	 * @param TableNode $resource
+	 *
+	 * @return void
+	 */
+	public function theActivityNumberShouldHaveAMessageAboutBatchRestoreInTheActivityPage($index, TableNode $resource) {
+		$latestActivityMessage = $this->activityPage->getActivityMessageOfIndex($index - 1);
+		$resource_array = [];
+		foreach ($resource->getHash() as $row) {
+			// checks if the file/folder name is present in the latest activity message log
+			if (\strpos($latestActivityMessage, $row['entry'])) {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIActivityContext.php]: The message might contain the entry multiple times, and the code will pass.
e.g. `You restored lorem.txt,  strängé नेपाली folder, lorem.txt and 'single'quotes
`substr_count()` can help.

```
			$entryCount = \substr_count($latestActivityMessage, $row['entry']);
			if ($entryCount === 1) {
				\array_push($resource_array, $row['entry']);
			} elseif ($entryCount < 1) {
				throw new Exception(sprintf("%s entry not present in the log", $row['entry']));
			} else {
				throw new Exception(sprintf("%s entry present %d times in the log", $row['entry'], $entryCount));
			}
```
Then we will know that each string occurs exactly once. And then the regex below will make sure about the format.

**Pull request merged**